File: src/main/java/sortingalgorithms/insertionsort/java/InsertionSort.java
package sortingalgorithms.insertionsort.java;

import org.jetbrains.annotations.NotNull;
import sortingalgorithms.SortingAlgorithm;

public class InsertionSort extends SortingAlgorithm {
    @Override
    public void sort(@NotNull Integer[] input) {
        int length = input.length;
        for (int i = 1; i < length; i++) {
            int current = input[i];
            int j = i - 1;
            while (j >= 0 && input[j] > current) {
                input[j + 1] = input[j];
                j--;
            }
            input[j + 1] = current;
        }
    }
}

File: src/main/java/sortingalgorithms/insertionsort/kotlin/InsertionSort.kt
package sortingalgorithms.insertionsort.kotlin

import sortingalgorithms.SortingAlgorithm

class InsertionSort : SortingAlgorithm() {
    override fun sort(input: IntArray) {
        var length = input.size
        for (i in 1 until length) {
            var current = input[i]
            var j = i - 1
            while (j >= 0 && input[j] > current) {
                input[j + 1] = input[j]
                j--
            }
            input[j + 1] = current
        }
    }
}